openapi2beans/pkg/generator/schema2package_test.go

```diff
@@ -0,0 +1,452 @@
+	}
+
+	// Data members generated ok
+	for _, dataMember := range class.DataMembers {
+		exists := false
+		var comparisonSchemaProperty *Property 
+		var expectedName string
+		comparisonSchemaProperty, exists = schemaType.properties[schemaPath+"/"+dataMember.Name]
+		expectedName = comparisonSchemaProperty.name
+		
+		assert.True(t, exists)
+		assert.Equal(t, expectedName, dataMember.Name)
+		expectedType := getExpectedType(comparisonSchemaProperty)
+		assert.Equal(t, expectedType, dataMember.MemberType)
+	}
+
+	// Constant data members generated ok
+	for _, constDataMember := range class.ConstantDataMembers {
+		exists := false
+		var comparisonSchemaProperty *Property 
+		var expectedName string
+		assert.NotEmpty(t, constDataMember.ConstantVal)
+		for _, prop := range schemaType.properties {
+			if convertToConstName(prop.name) == constDataMember.Name {
+				expectedName = convertToConstName(prop.name)
+				comparisonSchemaProperty = prop
+				exists = true
+				break
+			}
+		}
+		
+		assert.True(t, exists)
+		assert.Equal(t, expectedName, constDataMember.Name)
+		expectedType := getExpectedType(comparisonSchemaProperty)
+		assert.Equal(t, expectedType, constDataMember.MemberType)
+		assert.True(t, comparisonSchemaProperty.IsConstant())
+		posVal := possibleValuesToEnumValues(comparisonSchemaProperty.possibleValues)
+		assert.Equal(t, 1, len(posVal))
+		assert.Equal(t, convertConstValueToJavaReadable(posVal[0], comparisonSchemaProperty.typeName), constDataMember.ConstantVal)
+	}
+
+	// Required data members generated ok
+	requiredPropertiesVisited := 0
+	for _, requiredMember := range class.RequiredMembers {
+		comparisonSchemaProperty, exists := schemaType.properties[schemaPath+"/"+requiredMember.DataMember.Name]
+		assert.True(t, exists)
+		requiredPropertiesVisited += 1
+		expectedName := comparisonSchemaProperty.name
+		assert.Equal(t, expectedName, requiredMember.DataMember.Name)
+		expectedType := getExpectedType(comparisonSchemaProperty)
+		assert.Equal(t, requiredMember.DataMember.MemberType, expectedType)
+		assert.True(t, comparisonSchemaProperty.IsSetInConstructor())
+	}
+	assert.Equal(t, numberOfRequiredProperties(schemaType.properties), requiredPropertiesVisited)
+}
+
+func numberOfRequiredProperties(properties map[string]*Property) int {
+	counter := 0
+	for _, prop := range properties {
+		if prop.IsSetInConstructor() {
+			counter += 1
+		}
+	}
+	return counter
+}
+
+func getExpectedType(schemaProp *Property) string {
+	expectedType := ""
+	if schemaProp.typeName == "string" {
+		expectedType = "String"
+	} else if schemaProp.typeName == "integer" {
+		expectedType = "int"
+	} else if schemaProp.typeName == "number" {
+		expectedType = "double"
+	} else if schemaProp.typeName == "object" || schemaProp.IsEnum(){
+		expectedType = schemaProp.resolvedType.name
+	} else {
+		expectedType = schemaProp.typeName
+	}
+	if schemaProp.cardinality.max > 1 {
+		dimensions := schemaProp.cardinality.max / 128
```

Comment:
why divide by 128 ?

Reply:
the arbitrary limit of the java arrays I decided was 128, thinking about it, I'm going to make a constant for it first, but it would then make sense that the dimensions of the array would multiply according to the max limit.